New Lavaland Ruin: Crypt of the Necropolis!

Armhulen opened the pull request:
:cl: Armhulen coding, Keekenox spriting
add: New crypt ruin!!
/:cl:

Why? I've always wanted to contribute to lavaland, it's been getting a bit stale and I love ruins. It's a shame I haven't already! Here goes nothing...

Commit:
initial

Commit:
initial

Commit:
ayy lmao u wanna fuck a mapmerge

Armhulen commented:
sorry wip or something

Armhulen commented:
sprites comin soon

Shadowflame909 commented:
is this the lavaland blobbo practice rune? I hope it is...I wanna get some practice playing the now rare blob so I don't get ohko'd and decked by a fully powered 1 hour and 30 minute super crew.

Armhulen commented:
idk

Shadowflame909 commented:
I know kevinz loves his mechs, speed m e r g e.

davethwave commented:
Looks interesting.

Armhulen commented:
new commit soon:
final boss spawns a shadow of himself to finish the attack if it has to chase you
final boss has unique moves:
1. whirlwind, only attacks non-diagonals around himself
2. bladedance,  only attacks diagonals around himself
3. surge, fast attacks in the front, medium attacks in the middle and slow attack behind him
rooms themselves remade, removed lava for torches and added torches near the murals
new mural added

Time-Green commented on code/modules/mob/living/simple_animal/hostile/statue.dm:
```diff
@@ -2,7 +2,7 @@
 /mob/living/simple_animal/hostile/statue
 	name = "statue" // matches the name of the statue with the flesh-to-stone spell
-	desc = "An incredibly lifelike marble carving. Its eyes seem to follow you.." // same as an ordinary statue with the added "eye following you" description
+	desc = "An incredibly lifelike marble carving. It's eyes seem to follow you..." // same as an ordinary statue with the added "eye following you" description
```
"it's" means 'it is' or 'it has'
"its" was fine in this case

Armhulen replied:
It's eyes as in the statue's eyes

Time-Green replied:
possesive form is its
it's means it has or it is

Barhandar replied:
Both are valid for possessive form, the one without the apostrophe is better IMO for explicit purpose of distinguishing it from contraction of "it is" (contractions MUST have the apostrophe).

Armhulen replied:
we've been owned! he's right!

Time-Green replied:
god fucking dammit

Repukan commented:
Do these behave like normal statues or do they move normally?

Armhulen commented:
@Repukan not in the slightest do they work like statues

ExcessiveUseOfCobblestone commented:
You should consider new sprites.

ExcessiveUseOfCobblestone commented on code/modules/ruins/lavalandruin_code/tomb.dm:
```diff
@@ -0,0 +1,181 @@
+
+
+/mob/living/simple_animal/hostile/cryptguard/proc/awaken()
+	if(!istype(src, /mob/living/simple_animal/hostile/cryptguard))
```
can you explain this check

Armhulen replied:
yeah, only crypt guards have the woke proc so we really really wanna make sure it is a crypt guard

u-unless the proc being for cryptguards only is that...

ExcessiveUseOfCobblestone replied:
If only they have a proc when would this check ever fail? It would runtime at mob trying to call awaken because the mob in question doesn't have that proc.

If anything, the check would be above this (as in when it gets called).

Armhulen replied:
ok. thanks. i'll remove the check!

Armhulen commented:
@ExcessiveUseOfCobblestone alright... stick this in a sloppy joe. i'm bout to dump some mf sprites here and there

Armhulen commented:
![](https://i.imgur.com/LwUp3ZI.png)

![](https://i.imgur.com/eC0t88b.png)

ExcessiveUseOfCobblestone commented:
If you made these in 40 minutes i'm jealous.

Armhulen commented:
ME? no.

Commit:
some more work.

Bawhoppen commented:
Not sure if this is a joke or not, but incase it's not, then I vote no.
All it is, is just reused assets, and doesn't seem to actually add anything new, unique, or interesting.

Also you should fix the changelog.

Therapist1 commented:
It adds a mech. With a sword, and a neat boss fight too, which were made for this ruin specifically, so I don’t see what’s reused or boring about it.

ExcessiveUseOfCobblestone commented:
CAN I GET A LOOT SHOWCASE

Spaceinaba commented:
IF I CANT GET THAT SICK ASS STATUE ARMOR IM GONNA KILL YOU AND YOUR LITTLE FROG TOO

Armhulen commented on code/modules/ruins/lavalandruin_code/tomb.dm:
```diff
@@ -0,0 +1,254 @@
+	duration = 12
+	hit_damage = 50
+
+/obj/effect/temp_visual/attackwarn/Initialize(mapload, createdby, offset_x, offset_y)
```
reminder for myself to fix this

Armhulen commented on code/modules/ruins/lavalandruin_code/tomb.dm:
```diff
@@ -0,0 +1,254 @@
+	var/atktype = pick("cleave", "lunge")
+	visible_message("<span class='warning'>[src] winds up to [atktype] with it's greatsword...</span>", "<span class='notice'>You begin to wind up a [atktype]...</span>")
+	switch(atktype)
+		if("lunge")
```
and this

Commit:
yo bibbas i got a fix

Armhulen commented:
alright i got my idea of what iwll work but it isn't pretty

Commit:
test commit

Commit:
Merge branch 'patch-89' of https://github.com/Armhulen/tgstation into patch-89

Commit:
HE'S DONE IT

Commit:
adds energy blast and execute as an attack. energy blast wip.

kevinz000 commented:
Not necrodancer reference :-1:

Commit:
hiding under the floorboards i have finally found you

Armhulen commented:
@kevinz000 we already have the cursed heart you play fair now okay

Commit:
...a dangling participle... *BANG*

Commit:
JUMPING JESUS WORK MAN

Shadowflame909 commented:
Kevinz isn't that the hierophants job dood.

ExcessiveUseOfCobblestone commented:
you just lost your poster privileges sorry

Armhulen commented:
kevinz that's a week with no tables for you

Armhulen commented:
Cobby that wasn't my most recent commit keep in mind I'm a time traveler

I have uncommitted map I'll add tonight

Commit:
update

Wilchenx commented:
add a catgirl sarcophagus

Commit:
more code i guess

Commit:
necropolis wreckage yaey

Kromgar commented:
@Armhulen so is this a random spawn somewhere on the map or is it IN the actual necropolis? 

Also fix when?

Therapist1 commented:
also please I will indeed murder you if you do not give us that sick armor

imsxz commented on code/modules/ruins/lavalandruin_code/tomb.dm:
```diff
@@ -0,0 +1,363 @@
+	icon_state = "antikytherasword-sleeping"
+	icon = 'icons/mecha/mecha.dmi'
+	sentience_type = SENTIENCE_BOSS
+	loot = list(/obj/mecha/combat/stone/loaded)
```
Necropolis, not stone

imsxz commented on code/modules/ruins/lavalandruin_code/tomb.dm:
```diff
@@ -0,0 +1,363 @@
+	force = 40
+	wreckage = /obj/structure/mecha_wreckage/necropolis
+
+/obj/mecha/combat/stone/loaded/Initialize()
```
Necropolis, not stone. Additionally needs a loaded version defined somewhere first.

imsxz reviewed (commented):
frogcode

ChangelingRain commented on code/modules/ruins/lavalandruin_code/tomb.dm:
```diff
@@ -0,0 +1,363 @@
+				var/turf/Twhirl = get_step(src_turf, turn(dir_to_target, i))
+				new /obj/effect/temp_visual/attackwarn(Twhirl, src)
+		if("bdance") //attacks diagonals
+			var/static/list/bdance_angles = list(45, -45, 135, -135)
```
uh, you seem to be missing something here
also honestly I'd combine whirlwind and bladedance into one attack, randomly picking cardinal/diagonal

SpaceManiac commented:
@Armhulen Do me a favor - when you go to resolve merge conflicts on this, run `tools/hooks/install.bat` first before `git merge`, and ping me on coderbus if the automatic icon merger doesn't work. If it does work you should see output similar to this and the merge should succeed:

```
Merging icon: icons/obj/lighting.dmi
    'glowshroomf': deleted in right
    'glowcapf': deleted in right
    'shadowshroomf': deleted in right
    'crys': added in left
    'glowshroom': added in right
    'glowcap': added in right
    'shadowshroom': added in right
Merging icon: icons/obj/crates.dmi
    'sarcopen': added in left
    'sarc': added in left
    'coffin': added in right
    'coffinopen': added in right
Merging icon: icons/effects/effects.dmi
    'attackarea': added in left
    'explosion_indef': added in right
```

Commit:
small fixes
